packages/flutter/lib/src/painting/gradient.dart

```diff
@@ -16,22 +17,37 @@ class _ColorsAndStops {
   final List<double> stops;
 }
 
-_ColorsAndStops _interpolateColorsAndStops(List<Color> aColors, List<double> aStops, List<Color> bColors, List<double> bStops, double t) {
-  assert(aColors.length == bColors.length, 'Cannot interpolate between two gradients with a different number of colors.'); // TODO(ianh): remove limitation
-  assert((aStops == null && aColors.length == 2) || (aStops != null && aStops.length == aColors.length));
-  assert((bStops == null && bColors.length == 2) || (bStops != null && bStops.length == bColors.length));
-  final List<Color> interpolatedColors = <Color>[];
-  for (int i = 0; i < aColors.length; i += 1)
-    interpolatedColors.add(Color.lerp(aColors[i], bColors[i], t));
-  List<double> interpolatedStops;
-  if (aStops != null || bStops != null) {
-    aStops ??= const <double>[0.0, 1.0];
-    bStops ??= const <double>[0.0, 1.0];
-    assert(aStops.length == bStops.length);
-    interpolatedStops = <double>[];
-    for (int i = 0; i < aStops.length; i += 1)
-      interpolatedStops.add(ui.lerpDouble(aStops[i], bStops[i], t).clamp(0.0, 1.0));
-  }
+Color _sample(List<Color> colors, List<double> stops, double t) {
+  if (t <= stops.first)
+    return colors.first;
+  if (t >= stops.last)
+    return colors.last;
+  final int index = stops.lastIndexWhere((double s) => s <= t);
+  return Color.lerp(
+      colors[index], colors[index + 1],
+      (t - stops[index]) / (stops[index + 1] - stops[index])
+  );
+}
+
+_ColorsAndStops _interpolateColorsAndStops(
+    List<Color> aColors,
+    List<double> aStops,
+    List<Color> bColors,
+    List<double> bStops,
+    double t,
+) {
+  assert(aColors.length >= 2);
+  assert(bColors.length >= 2);
+  assert(aStops.length == aColors.length);
+  assert(bStops.length == bColors.length);
+  final SplayTreeSet<double> stops = SplayTreeSet<double>(
```

Comment:
Isn't comareTo the default if nothing is provided? Can these lines be changed to:

```dart
final SplayTreeSet<double> stops = SplayTreeSet<double>()
    ..addAll(aStops)
    ..addAll(bStops);
```

Reply:
That makes sense. Strange, somehow I missed that completely.